lib/es-models/checkin-card-item.js

```diff
@@ -0,0 +1,253 @@
+   */
+  holdingLocation () {
+    const holdingLocation = this.sierraHolding.location
+
+    if (!holdingLocation || !holdingLocation.code) return null
+    if (!nyplCoreLocations[holdingLocation.code]) {
+      logger.warn(`Holding location not recognized: ${holdingLocation.code}`)
+      return null
+    }
+
+    const id = `loc:${holdingLocation.code}`
+    const label = nyplCoreLocations[holdingLocation.code].label
+    return [{ id, label }]
+  }
+
+  holdingLocation_packed () {
+    return pack(this.holdingLocation())
+  }
+
+  /**
+   *  Get array of identifiers as entities
+   */
+  identifierV2 () {
+    return this._identifiers()
+  }
+
+  /**
+   *  Get shelf-mark from Holding record
+   */
+  shelfMark () {
+    const mappings = HoldingMappings.get('shelfMark', this.sierraHolding)
+    return this.sierraHolding.varFieldsMulti(mappings)
+      .map((varFieldMatch) => {
+        // Sierra seems to put these '|h' prefixes on callnumbers; strip 'em
+        return varFieldMatch.value.replace(/^\|h/, '')
+      })
+  }
+
+  shelfMark_sort () {
+    return this._sortableShelfMark()
+  }
+
+  /**
+   *  Get status by translating checkin card's status into an item status
+   */
+  status () {
+    const boxStatus = this.checkinCard.status.code
+    const translatedStatusCode = lookup('lookup-checkin-card-status-to-item-status')[boxStatus]
+    const status = nyplCoreStatuses[translatedStatusCode]
+    const id = 'status:' + status.id.split(':').pop()
+    return [
+      { id, label: status.label }
+    ]
+  }
+
+  status_packed () {
+    return pack(this.status())
+  }
+
+  /**
+   *  Get static checkin-card type
+   */
+  type () {
+    return [{ id: 'nypl:CheckinCardItem' }]
+  }
+
+  uri () {
+    return `i-h${this.sierraHolding.id}-${this.checkinCardIndex}`
+  }
+
+  /**
+   *  Get indexable volume range by parsing checkin card enumeration
+   */
+  volumeRange () {
+    if (!this.checkinCard.enumeration) return null
+    if (!this.checkinCard.enumeration.enumeration) return null
+
+    const volumeRanges = parseVolume(this.checkinCard.enumeration.enumeration)
+    // If no parsed ranges, return null:
+    if (!volumeRanges.length) return null
+
+    return volumeRanges.map(arrayToEsRangeObject)
+  }
+
+  /**
+   *  Get raw volume range by from checkin card enumeration
+   */
+  volumeRaw () {
+    if (!this.checkinCard.enumeration) return null
+
+    return [this.checkinCard.enumeration.enumeration]
+  }
+
+  /**
+   *  Get all relevant identifiers as entities
+   */
+  _identifiers () {
+    let identifiers = []
+
+    // Add shelfmark (call number):
+    if (this.shelfMark()) {
+      identifiers = identifiers.concat(
+        this.shelfMark().map((value) => ({ value, type: 'bf:ShelfMark' }))
+      )
+    }
+
+    return identifiers
+  }
+
+  /**
+   *  Given a SierraHolding instance, returns an array of EsCheckinCardItem
+   *  instances parsed from it
+   */
+  static fromSierraHolding (sierraHolding) {
```

Comment:
can this be called from the holding and not from within this class? its tricky to wrap my head around.